File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int>mpp;
        for(int i = 0; i < nums.size(); i++) {
            int compliment = target-nums[i];
            if(mpp.find(compliment) != mpp.end()) {
                return {mpp[compliment], i};
            }
            mpp[nums[i]] = i;
        }
        return {-1, -1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> mpp = new HashMap<>();

        for(int i = 0; i < nums.length; i++) {
            int reqNum = target - nums[i];

            if(mpp.containsKey(reqNum)) {
                return new int []{mpp.get(reqNum), i};
            }
            mpp.put(nums[i], i);
        }
        return new int []{};
    }
}